Feature/registerpage

RobertoWTBW opened the pull request.

CatinasDaria committed:
Created register page

RobertoWTBW committed:
Merge branch 'master' into feature/registerpage

RobertoWTBW committed:
Login page readded in registerpage branch

RobertoWTBW committed:
Register page is ready for meging

Done validators + resolved merging conflicts

CimpeanAndreea commented on lib/constants/constants.dart:
```diff
@@ -6,6 +6,12 @@ const gradientBaseGreen = 0xff2a9d8f;
 //PATHS
 
+const orangeLogoPath = "assets/bluelogo.png";
+const RegisterPageGradientPath = "assets/registerpagegradient.png";
```
Camel case this too

CimpeanAndreea commented on lib/loginpage/loginpage_form.dart:
```diff
@@ -37,23 +37,31 @@ class _MyWidgetState extends State<LoginPageForm> {
       }
+      else {
+        setState(() {
+          _outlineColorPassword = Color(0xffb9b9b9);
```
Define this color as a constant somewhere and use it below too

CimpeanAndreea commented on lib/registerpage/registerpage.dart:
```diff
@@ -0,0 +1,40 @@
+            width: double.infinity,
+            decoration: const BoxDecoration(
+                image: DecorationImage(
+                    image: AssetImage("assets/registerpagegradient.png"),
```
Use the constant that you defined for the path

CimpeanAndreea commented on lib/registerpage/registerpage.dart:
```diff
@@ -0,0 +1,40 @@
+                            child: Container(
+                                height: double.infinity,
+                                width: double.infinity,
+                                color: Color.fromARGB(255, 42, 157, 143))),
```
Define a constant for the color, you might need it somewhere else

CimpeanAndreea commented on lib/registerpage/registerpage_form.dart:
```diff
@@ -0,0 +1,250 @@
+    }
+    else {
+      setState(() {
+        _outlineBorderColorPassword = Color(0xffb9b9b9);
```
See comment above for the color

CimpeanAndreea commented on lib/registerpage/registerpage_form.dart:
```diff
@@ -0,0 +1,250 @@
+            Positioned(
+              top: widget.screenHeight * 0.07,
+              child: DefaultTextStyle(
+                style: TextStyle(color: Color.fromARGB(255, 27, 21, 18), fontSize: widget.screenWidth * 0.0275, fontWeight: FontWeight.bold),
```
Constant for the color, below too

CimpeanAndreea commented on lib/registerpage/registerpage_image.dart:
```diff
@@ -0,0 +1,23 @@
+        child: DecoratedBox(
+          decoration: BoxDecoration(
+            image: DecorationImage(
+              image: AssetImage("assets/registerpageimage.png"),
```
Use the constant you defined for this

CimpeanAndreea commented on lib/registerpage/registerpage_logo.dart:
```diff
@@ -0,0 +1,20 @@
+      right: 0,
+      top: screenHeight * 0.016,
+      child: Image.asset(
+        "assets/bluelogo.png",
```
The constant you defined :D

CimpeanAndreea commented on pubspec.yaml:
```diff
@@ -58,8 +58,8 @@ flutter:
 
   assets:
-    - assets/bluegradient.png
-    - assets/loginpageimage.png
```
Are we using the same image for register and login? Just want to make sure these lines were not deleted by mistake. And if not, rename it, since it's not only used in register. Also is this gradient the same in the entire app? Then it definitely should not have "register" in its name.

CimpeanAndreea commented on lib/constants/constants.dart:
```diff
@@ -6,6 +6,12 @@ const gradientBaseGreen = 0xff2a9d8f;
\ No newline at end of file
+const loginPageImagePath = "assets/loginpageimage.png";
+
+const pattern = r'^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[!@#\$&*~]).{8,}$';
```
Rename to be more suggestive. You might have a pattern for something else in the future.

CimpeanAndreea commented on lib/registerpage/registerpage_form.dart:
```diff
@@ -0,0 +1,250 @@
+
+  final mailController = TextEditingController();
+  final passwordController = TextEditingController();
+  final secondPasswordController = TextEditingController();
```
Is this supposed to be for the "confirm password" field? Rename it if so.

CimpeanAndreea commented on lib/registerpage/registerpage_form.dart:
```diff
@@ -0,0 +1,250 @@
+  final passwordController = TextEditingController();
+  final secondPasswordController = TextEditingController();
+
+  var _outlineBorderColorMail = Color(0xffb9b9b9);
```
Constants for colors, everywhere 😉 If you were to change the palette and also this particular color across the entire app, by having a constant you would have to change it in only 1 place and you will be sure it will change it everywhere.

CimpeanAndreea commented on lib/registerpage/registerpage_form.dart:
```diff
@@ -0,0 +1,250 @@
+                      color: Color.fromARGB(255, 42, 157, 143),
+                    )
+                  ),
+                  onPressed: () {},
```
Should you navigate here to the login page? I don't see anymore a point from where you access the login page.

CimpeanAndreea commented on lib/registerpage/registerpage_image.dart:
```diff
@@ -0,0 +1,23 @@
+  @override
+  Widget build(BuildContext context) {
+    return Expanded(
+      flex: 90, //avem valori asa de mari aici pentru a face bara colorata din mijloc mica
```
Remove the comment. Keep only important ones and translate them. Make them "TODO" comments if necessary.

RobertoWTBW committed:
Code review assessed

The code review was assessed, except for one comment. We do not have a button to navigate between login and register page

RobertoWTBW committed:
Login - Register navigation

Implemented way of navigating between login and register page

RobertoWTBW committed:
Moved the sign up button in the register page

Moved the sign up button in the register page
